### PR TITLE
Auto code surveys

### PR DESCRIPTION
**Pull request by IsaackMwenda**
Hi @as2388 have added the auto code surveys step. Pending this https://github.com/AfricasVoices/Test-REDDS/pull/11 . Thanks

### CODE REVIEW

**Comment by as2388** [comment]
@IsaackMwenda please merge master into this branch

**Comment by IsaackMwenda** [comment]
> @IsaackMwenda please merge master into this branch

File: project_test/auto_code_surveys.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+class AutoCodeSurveys(object):
+    SENT_ON_KEY = "sent_on"
+
+    @classmethod
+    def auto_code_surveys(cls, user, data, phone_uuid_table, coda_output_dir):
+        # Label missing data
+        for td in data:
+            missing_dict = dict()
+            for plan in PipelineConfiguration.SURVEY_CODING_PLANS:
+                if td.get(plan.raw_field, "") =="":
```
Comment (as2388) [inline project_test/auto_code_surveys.py]:
```suggestion
                if td.get(plan.raw_field, "") == "":
```

File: test_pipeline.py [hunk 2]
```diff
@@ -122,6 +123,9 @@
 
     print("Auto coding Messages...")
     data = AutoCodeShowMessages.auto_code_show_messages(user, data, icr_output_dir, coded_dir_path)
+    
+    print("Auto coding surveys...")
+    data =AutoCodeSurveys.auto_code_surveys(user, data, phone_number_uuid_table_path, coded_dir_path)
```
Comment (as2388) [inline test_pipeline.py]:
```suggestion
    data = AutoCodeSurveys.auto_code_surveys(user, data, phone_number_uuid_table_path, coded_dir_path)
```

File: project_test/auto_code_surveys.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        # Auto-code remaining data
+        for plan in PipelineConfiguration.SURVEY_CODING_PLANS:
+            if plan.cleaner is not None:
+                CleaningUtils.apply_cleaner_to_traced_data_iterable(user, data, plan.raw_field, plan.coded_field,
+                                                                    plan.cleaner, plan.code_scheme)
+                                                                    
+        # Output survey answers to coda for manual verification + coding
+        IOUtils.ensure_dirs_exist(coda_output_dir)
+        for plan in PipelineConfiguration.SURVEY_CODING_PLANS:
+
```
Comment (as2388) [inline project_test/auto_code_surveys.py]:
```suggestion
```

**Review by as2388 — APPROVED** [review]

**Pull request merged**